src/io/forsta/ccsm/messaging/ForstaMessageManager.java

```diff
@@ -169,16 +169,17 @@ private static ForstaMessage parseMessageBody(JSONObject jsonBody) throws Invali
         // This is a special case. Message type is CONTENT,
         // but processing like a control message because
         // we don't want to save these messages to the message table.
-        if (data.has("vote")) {
-          if (data.has("messageRef")) {
-            String messageId = data.getString("messageRef");
-            // Override message type.
+
+        if (data.has("messageRef")) {
+          String messageId = data.getString("messageRef");
```

Comment:
What happens if messageRef is null or otherwise not a string?  Exception ?

Reply:
Jason, my PR includes everything for the messageRef, including a new field in the message db to store the message ref so that we can search on that field. Will you remove these changes from this PR and just have the Provision fix in this one.